rviz_default_plugins/src/rviz_default_plugins/displays/depth_cloud/depth_cloud_display.cpp

```
@@ -0,0 +1,602 @@
+    new rviz_common::properties::BoolProperty(
+    "Occlusion Compensation", false,
+    "Keep points alive after they have been occluded by a closer point. "
+    "Points are  removed after a timeout or when the camera frame moves.",
+    this, SLOT(updateUseOcclusionCompensation()));
+
+  occlusion_shadow_timeout_property_ = new rviz_common::properties::FloatProperty(
+    "Occlusion Time-Out", 30.0f,
+    "Amount of seconds before removing occluded points from the depth cloud",
+    use_occlusion_compensation_property_, SLOT(updateOcclusionTimeOut()), this);
+}
+
+void DepthCloudDisplay::onInitialize()
+{
+  auto rviz_ros_node_ = context_->getRosNodeAbstraction();
+
+  depthmap_it_.reset(new image_transport::ImageTransport(rviz_ros_node_.lock()->get_raw_node()));
+  rgb_it_.reset(new image_transport::ImageTransport(rviz_ros_node_.lock()->get_raw_node()));
+
+  // Instantiate PointCloudCommon class for displaying point clouds
+  pointcloud_common_ = new PointCloudCommon(this);
+
+  updateUseAutoSize();
+  updateUseOcclusionCompensation();
+
+  // Scan for available transport plugins
+  scanForTransportSubscriberPlugins();
+
+  pointcloud_common_->initialize(context_, scene_node_);
+  pointcloud_common_->xyz_transformer_property_->hide();
+
+  depth_topic_property_->initialize(context_->getRosNodeAbstraction());
+  color_topic_property_->initialize(context_->getRosNodeAbstraction());
+}
+
+DepthCloudDisplay::~DepthCloudDisplay()
+{
+  if (initialized()) {
+    unsubscribe();
+    delete pointcloud_common_;
+  }
+
+  if (ml_depth_data_) {
+    delete ml_depth_data_;
+  }
+}
+
+void DepthCloudDisplay::setTopic(const QString & topic, const QString & datatype)
+{
+  // Copied from ImageDisplayBase::setTopic()
+  if (datatype == "sensor_msgs::msgs::Image") {
+    depth_transport_property_->setStdString("raw");
+    depth_topic_property_->setString(topic);
+  } else {
+    int index = topic.lastIndexOf("/");
+    if (index == -1) {
+      return;
```

We definitely should have this check, but it also silently seems to silently fail.  Should we set an error message somewhere the user can see it?

---

This comment is still open.

---

This method is filling some of the elements in the GUI, I don't think we should return something different, We are looking specific topics and this might generate unnecessary noise in the CMD.